### PR TITLE
Add factory for reading from generic molecule file. 

### LINKED ISSUE #459: Make a factory to make Ligands from a generic molecule file.
Make a factory to make Ligands from a generic molecule file.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #459

### CODE REVIEW

**Comment by hmacdope** [comment]
@kaminow could I grab a review here?

**Review by kaminow — CHANGES_REQUESTED** [review]
couple tiny things but great work as always!

File: asapdiscovery-data/asapdiscovery/data/tests/test_molfile_factory_schema_v2.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    return fetch_test_file("Mpro_combined_labeled.smi")
+
+
+def test_molfile_factory_sdf(sdf_file):
+    molfile = MolFileFactory.from_file(filename=sdf_file)
+    assert len(molfile.ligands) == 576
+
+
+def test_molfile_factory_smi(smi_file):
+    molfile = MolFileFactory.from_file(filename=smi_file)
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply asapdiscovery-data/asapdiscovery/data/tests/test_molfile_factory_schema_v2.py:22]: any file format that ti can recognise easily including mol, csv, sdf etc, pretty handy.

File: asapdiscovery-data/asapdiscovery/data/schema_v2/molfile.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        retcode = ifs.open(str(filename))
+        if not retcode:
+            raise ValueError(f"Could not open {filename}")
+
+        mollist = []
+        for mol in ifs.GetOEGraphMols():
+            mollist.append(oechem.OEGraphMol(mol))
+
+        ligands = []
+        for i, mol in enumerate(mollist):
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply asapdiscovery-data/asapdiscovery/data/schema_v2/molfile.py]: great point, fixed.

**Review by kaminow — APPROVED** [review]
looks great!

**Pull request merged**